service-proxy.js

```diff
@@ -918,6 +961,7 @@ function ensurePeerDisconnected(serviceName, peer, reason, now) {
 
     var peerServices = self.connectedPeerServices[peer.hostPort];
     if (!peerServices || isObjectEmpty(peerServices)) {
+        delete self.peersToConnect[peer.hostPort];
```

Comment:
Good question, I just updated the connectSinglePeer implementation to:
- stop if there's a prune note
- not create the peer if it doesn't exist
- stop if the peer, if it still exists, is draining

thoughts?